### PR TITLE
For - While Loop refactored

### LINKED ISSUE #344: remove all 'for' loops from source code
by the ANSI standard the first field of `for` ends up not being used to instantiate variables, besides using while would be something simpler and more readable to maintain.

### before ###

```c
int i = 0;

for (; i < sizeof(any); i++) {
   /** code **/
}
```

### after ###

```c
int i = 0;

while(i < sizeof(any)) {
   /** code **/
   i += 1;
}
```

### PR DESCRIPTION
**Pull request by Yugal41735**
# Description

For-while loop refactored as mentioned in issue

closes #344 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: src/ds_label_hash.c [hunk 1]
```diff
@@ -71,8 +71,9 @@ void ds_label_hash_insert(struct app_3bc_s* const app, label_3bc_t label)
     }
 
     last_node = app->program.label_table[hash];
-    for (; last_node->next != NULL; last_node = last_node->next)
+    while ( last_node->next != NULL)
         ;
+        last_node = last_node->next;
```
Comment (RodrigoDornelles) [inline src/ds_label_hash.c:76]: to maintain the same behavior change to:
```c
while (last_node->next != NULL) {
      last_node = last_node->next;
 }
 ```

**Review by RodrigoDornelles — CHANGES_REQUESTED** [review]
There are some changes to be made!

please I just discovered yesterday that `++i` has a performance gain in 6502 family, would like to see this in the same PR.

ref: https://cc65.github.io/doc/coding.html item 9

**Comment by Yugal41735** [comment]
Done, the necessary changes. You could review it again.

**Comment by Yugal41735** [comment]
Done, the necessary changes. You could review it again.  @RodrigoDornelles

**Review by RodrigoDornelles — APPROVED** [review]

File: src/driver_custom.c [hunk 2]
```diff
@@ -65,8 +65,10 @@ void driver_custom_set(struct app_3bc_s* const app, cpumode_3bc_t cpu_mode,
         }
 
         /** clean wild pointers **/
-        for (; last_func <= atual_func; last_func++) {
+        while (last_func <= actual_func)
+        {
             new_array[last_func] = NULL;
+            ++last_func;
```
Comment (RodrigoDornelles) [inline src/driver_custom.c:71]: ### error occurred ###
```
In file included from programs/../src/3bc.h:133,
                 from programs/tests.c:2:
programs/../src/driver_custom.c: In function ‘void driver_custom_set(app_3bc_s*, cpumode_3bc_t, register_3bc_t, function_3bc_t)’:
programs/../src/driver_custom.c:6[8](https://github.com/RodrigoDornelles/3bc-lang/actions/runs/3274913903/jobs/5389836982#step:3:9):2[9](https://github.com/RodrigoDornelles/3bc-lang/actions/runs/3274913903/jobs/5389836982#step:3:10): error: ‘actual_func’ was not declared in this scope; did you mean ‘atual_func’?
   68 |         while (last_func <= actual_func)
      |                             ^~~~~~~~~~~
      |                             atual_func
make: *** [Makefile:86: tests-full] Error 1
Error: Process completed with exit code 2.
```

### to solve ###

its my typo wrong, but the **actual** term is still not so correct, replace the `actual_func` and `current_func` in place use `current_func`

File: src/interpreter_tokens.c [hunk 3]
```diff
@@ -65,16 +65,16 @@ bool interpreter_tokens(
 
     do {
         /** search for the beginning of the column **/
-        for (; strchr("\t. ", *pointer) != NULL && pointer[0] != '\0';
-             pointer++)
-            ;
+        while (strchr("\t. ", *pointer) != NULL && pointer[0] != '\0'){
```
Comment (RodrigoDornelles) [inline src/interpreter_tokens.c:68]: format: `while (strchr("\t. ", *pointer) != NULL && pointer[0] != '\0') {`

File: src/interpreter_tokens.c [hunk 4]
```diff
@@ -65,16 +65,16 @@ bool interpreter_tokens(
...
+            ++pointer;
+        }
 
         /** skip comments **/
         if (strchr("#;", *pointer) != NULL) {
-            for (;
-                 pointer[0] != '\n' && pointer[0] != '\0' && pointer[0] != EOF;
-                 pointer++)
-                ;
+            while (pointer[0] != '\n' && pointer[0] != '\0' && pointer[0] != EOF){
```
Comment (RodrigoDornelles) [inline src/interpreter_tokens.c:74]: format: `while (pointer[0] != '\n' && pointer[0] != '\0' && pointer[0] != EOF) {`

File: src/driver_custom.c (hunk 2 above)
Comment (RodrigoDornelles) [reply src/driver_custom.c:71]: don't forget to take care of the indentation, you can use clang-format to check.

File: src/ds_label_hash.c [hunk 5]
```diff
@@ -71,8 +71,9 @@ void ds_label_hash_insert(struct app_3bc_s* const app, label_3bc_t label)
     }
 
     last_node = app->program.label_table[hash];
-    for (; last_node->next != NULL; last_node = last_node->next)
-        ;
+    while ( last_node->next != NULL) {
```
Comment (RodrigoDornelles) [inline src/ds_label_hash.c:74]: format: `while (last_node->next != NULL) {`

**Review by RodrigoDornelles — CHANGES_REQUESTED** [review]

File: src/interpreter_tokens.c [hunk 6]
```diff
@@ -65,16 +65,16 @@ bool interpreter_tokens(
...
         /** skip comments **/
         if (strchr("#;", *pointer) != NULL) {
-            for (;
-                 pointer[0] != '\n' && pointer[0] != '\0' && pointer[0] != EOF;
-                 pointer++)
-                ;
+            while (pointer[0] != '\n' && pointer[0] != '\0' && pointer[0] != EOF) {
+                ++pointer;
+            }
+
```
Comment (RodrigoDornelles) [inline src/interpreter_tokens.c:77]: problem is it must be within the 80 characters of column.
```c
            while (
                pointer[0] != '\n' && pointer[0] != '\0' && pointer[0] != EOF) {
                ++pointer;
            }

            break;
        }
```

**Review by RodrigoDornelles — CHANGES_REQUESTED** [review]

**Comment by RodrigoDornelles** [comment]
it's been so long since the indentation errors, you can try again, but if it doesn't work I'll close the pullrequest without the merge. @Yugal41735

**Comment by Yugal41735** [comment]
I will try again and let you know.

**Comment by Yugal41735** [comment]
If this does not work, I could try again, and do you know of any online tool, for checking clang-format, bcz due to some internal errors, I am having difficulty in checking clang-format locally.

**Review by RodrigoDornelles — APPROVED** [review]

**Pull request merged**